hail/python/hail/matrixtable.py

```diff
@@ -2466,6 +2466,21 @@ def checkpoint(self, output: str, overwrite: bool = False, stage_locally: bool =
         >>> dataset = dataset.checkpoint('output/dataset_checkpoint.mt')
 
         """
+        if _codec_spec is None:
```

Comment:
this requires documentation

Reply:
It's a hidden argument, I don't think it should be documented.

Reply:
checkpoint now has different semantics from `write` -- the files are bigger. That's the piece we should document.